haystack/document_stores/dc.py

```diff
@@ -0,0 +1,341 @@
+        Get documents from the document store. Under-the-hood, documents are fetched in batches from the
+        document store and yielded as individual documents. This method can be used to iteratively process
+        a large number of documents without having to load all documents in memory.
+
+        :param index: Name of the index to get the documents from. If None, the
+                      DocumentStore's default index (self.index) will be used.
+        :param filters: Optional filters to narrow down the documents to return.
+                        Example: {"name": ["some", "more"], "category": ["only_one"]}
+        :param return_embedding: Whether to return the document embeddings.
+        :param batch_size: When working with large number of documents, batching can help reduce memory footprint.
+        :param headers: Custom HTTP headers to pass to document store client if supported (e.g. {'Authorization': 'Basic YWRtaW46cm9vdA=='} for basic authentication)
+        """
+        if batch_size != 10_000:
+            raise ValueError("DCDocumentStore does not support batching")
+
+        if return_embedding is not None and return_embedding != self.return_embedding:
+            raise ValueError("DCDocumentStore does not support dynamic return_embeddings values")
+
+        if index is None:
+            index = self.index
+        
+        if return_embedding is None:
+            return_embedding = self.return_embedding   
+
+        body = {
+            "return_embedding": return_embedding,
+            "filters": filters
+        }
+
+        body = self._remove_null_values(body)
+        url = f"{self._get_index_endpoint(index)}/documents-stream"
+        response = requests.post(url=url, json=body, stream=True, headers=headers, auth=BearerAuth(self.api_key))
+        if response.status_code != 200:
+            raise Exception(f"An error occured while loading documents: HTTP {response.status_code} - {response.reason}\n{response.content.decode()}")
+        
+        for raw_doc in response.iter_lines():
+            dict_doc = json.loads(raw_doc.decode('utf-8'))
+            yield Document.from_dict(dict_doc)
+
+    def get_document_by_id(self, id: str, index: Optional[str] = None, headers: Optional[Dict[str, str]] = None) -> Optional[Document]:
+        if index is None:
+            index = self.index
+        
+        query_params = {
+            "return_embedding": self.return_embedding
+        }
+        url = f"{self._get_index_endpoint(index)}/documents/{id}"
+        response = requests.get(url=url, headers=headers, auth=BearerAuth(self.api_key), params=query_params)
+
+        doc: Optional[Document] = None
+        if response.status_code == 200:
+            doc_dict = response.json()
+            doc = Document.from_dict(doc_dict)
+        else:
+            logger.warning(f"Document {id} could not be fetched from Deepset Cloud: HTTP {response.status_code} - {response.reason}\n{response.content.decode()}")
```

Comment:
No, as this method is also used by `get_documents_by_id`, this would break the whole "request" after say 100 docs had been successfully fetched already. Other documentstores like ElasticsearchDocumentStore behave the same way.